### PR TITLE
udev: fix by-id symlinks

### LINKED ISSUE #25238: udev: missing /dev/serial/by-id/usb-*
After the v252-rc3 → v252 uprade, udev no longer creates `serial/by-id` symlinks for my `/dev/ttyUSB*` devices.

See the "udevadm info /dev/ttyUSB0" dumps for both versions:
* [v252-rc3](https://bugs.debian.org/cgi-bin/bugreport.cgi?att=1;bug=1023311;filename=log.252~rc3-2;msg=3)
* [v252](https://bugs.debian.org/cgi-bin/bugreport.cgi?att=2;bug=1023311;filename=log.252-2;msg=3)

I guess 479da1107a0d4e2f is the culprit? Looks like `usb_id` no longer sets `ID_SERIAL`, only `ID_USB_SERIAL`, but `60-serial.rules` uses only the former.

### PR DESCRIPTION
**Pull request by yuwata**
Fixes a bug introduced by 479da1107a0d4e2f7ef5cd938512b87a0e45f180.

Fixes #25238.

### CODE REVIEW

**Comment by yuwata** [comment]
The first commit needs to be backported to v252-stable.

**Comment by jwilk** [comment]
https://github.com/systemd/systemd/compare/6c83054c0133eb53245e479d71589dceff76cf74...5286da064c97d2ac934cb301066aaa8605a3c8f9 fixes it for me.
Thanks.

File: rules.d/60-serial.rules [hunk 1]
```diff
@@ -3,9 +3,10 @@
 ACTION=="remove", GOTO="serial_end"
 SUBSYSTEM!="tty", GOTO="serial_end"
 
-SUBSYSTEMS=="pci", ENV{ID_BUS}="pci", ENV{ID_VENDOR_ID}="$attr{vendor}", ENV{ID_MODEL_ID}="$attr{device}"
-SUBSYSTEMS=="pci", IMPORT{builtin}="hwdb --subsystem=pci"
 SUBSYSTEMS=="usb", IMPORT{builtin}="usb_id", IMPORT{builtin}="hwdb --subsystem=usb"
+SUBSYSTEMS=="pci", ENV{ID_BUS}=="", ENV{ID_BUS}="pci", \
```
Comment (bluca) [inline rules.d/60-serial.rules:7]: why is `ENV{ID_BUS}` specified twice?

Comment (yuwata) [reply rules.d/60-serial.rules:7]: The first one is not an assignment, but checking if it is empty.

**Review by bluca — APPROVED** [review]

**Comment by keszybz** [comment]
The first commit was already included in v252.1.

**Comment by m3cinc** [comment]
Apparently the fis is not sufficient as the scripts are still not working in Debian 6.1.21-v7l+ #1642 SMP Mon Apr  3 17:22:30 BST 2023 armv7l, the current distributed version:
ls: cannot access '/dev/serial/by-id/': No such file or directory
Hope you can resolve as this impairs klipper functionality today!

**Comment by yuwata** [comment]
@m3cinc Please open a new issue page with `udevadm info DEVICE_PATH`, `udevadm test DEVICE_PATH`, and the full contents of 60-serial.rules.
Otherwise, comments on already merged PR will be easily forgotten.

**Comment by m3cinc** [comment]
Yu,
done.
I really tried 32bit and 64bit OS with 6+ kernel... same issue, so I posted
with all info as you recommended.
Thanks for the return.
Hope this can be solved soon.
Marc

**Comment by catchthemonster** [comment]
Folks, this does not work, the dongle for zwave is disappearing with 60-serial.rules
what version will fix this?
this really suck

**Comment by yuwata** [comment]
@catchthemonster 
The commit is backported to v252 as https://github.com/systemd/systemd-stable/commit/dafd73367b4a30e274eb7e06bd610648aab1cc15, which is included in v252.1.

Please also see #28509.

**Comment by yuwata** [comment]
@catchthemonster 
Anyway, as I said in the above https://github.com/systemd/systemd/pull/25246#issuecomment-1722512348, please open a new issue with detailed info.

**Pull request merged**